### PR TITLE
Fixed #34036 -- Corrected low link color contrast in admin light-theme. 

### PR DESCRIPTION
**Pull request by nimra200**
This PR is for [improving color contrast in the admin light theme](https://code.djangoproject.com/ticket/34036). Google Chrome has a useful tool for website readability called [CSS Overview](https://developer.chrome.com/docs/devtools/accessibility/contrast/#overview-contrast). The tool indicates the color contrast ratios now meet AA guidelines of [WebAIM ](https://webaim.org/standards/wcag/#current).
Screen Shot 2022-11-07 at 3 43 34 PM
@carltongibson - Would appreciate if you could look at this!

### CODE REVIEW

**Comment by felixxm** [comment]
@nimra200 Thanks for this patch :+1: Please revert all unrelated reformatting as it makes review more difficult.

File: django/contrib/admin/static/admin/css/base.css [hunk 1]
```diff
@@ -1128,11 +1230,12 @@ a.deletelink:focus, a.deletelink:hover {
...
+.paginator a:focus,
+.paginator a:hover {
     color: white;
     background: var(--link-hover-color);
 }
 
 .base-svgs {
     display: none;
-}
+}
```
Comment (felixxm) [inline django/contrib/admin/static/admin/css/base.css:1241]: Please restore the newline character.

**Comment by felixxm** [comment]
@nimra200 Do you have time to keep working on this?

File: django/contrib/admin/static/admin/css/base.css [hunk 2]
```diff
@@ -104,7 +104,7 @@ body {
 /* LINKS */
 
 a:link, a:visited {
-    color: var(--link-fg);
+    color: var(--link-hover-color);
```
Comment (felixxm) [inline django/contrib/admin/static/admin/css/base.css:107]: It's covered by #16101 and ticket-34038. I think we should make `--link-fg` color darker to meet AA, not change to `--link-hover-color` (see [comment](https://github.com/django/django/pull/16101/files#r1022588796)).

File: django/contrib/admin/static/admin/css/base.css [hunk 3]
```diff
@@ -24,7 +24,7 @@ html[data-theme="light"],
     --breadcrumbs-link-fg: var(--body-bg);
     --breadcrumbs-bg: var(--primary);
 
-    --link-fg: #447e9b;
+    --link-fg: #417893;
```
Comment (felixxm) [inline django/contrib/admin/static/admin/css/base.css:27]: Moved to the #16378.

File: django/contrib/admin/static/admin/css/base.css [hunk 4]
```diff
@@ -774,7 +774,7 @@ a.deletelink:focus, a.deletelink:hover {
     float: left;
     padding: 3px 12px;
     background: var(--object-tools-bg);
-    color: var(--object-tools-fg);
```
[the remark this answers is not in the document]
Comment (nimra200) [reply django/contrib/admin/static/admin/css/base.css]: Thanks for the suggestion. I've made the requested changes.

File: django/contrib/admin/static/admin/css/base.css [hunk 5]
```diff
@@ -704,12 +704,12 @@ div.breadcrumbs {
     background: var(--breadcrumbs-bg);
     padding: 10px 40px;
     border: none;
-    color: var(--breadcrumbs-fg);
+    color: var(--body-fg);
     text-align: left;
 }
 
 div.breadcrumbs a {
-    color: var(--breadcrumbs-link-fg);
```
[the remark this answers is not in the document]
Comment (nimra200) [reply django/contrib/admin/static/admin/css/base.css]: I've made the requested changes.

File: django/contrib/admin/static/admin/css/base.css [hunk 6]
```diff
@@ -540,7 +540,7 @@ a.button {
 .button.default, input[type=submit].default, .submit-row input.default {
     border: none;
     font-weight: 400;
-    background: var(--default-button-bg);
```
[the remark this answers is not in the document]
Comment (nimra200) [reply django/contrib/admin/static/admin/css/base.css]: I've made the requested changes.

File: django/contrib/admin/static/admin/css/base.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (nimra200) [reply django/contrib/admin/static/admin/css/base.css]: @felixxm

File: django/contrib/admin/static/admin/css/widgets.css [hunk 7]
```diff
@@ -41,7 +41,7 @@
 }
 
 .selector-chosen h2 {
-    background: var(--primary);
+    background: var(--secondary);
```
Comment (thibaudcolas) [inline django/contrib/admin/static/admin/css/widgets.css:44]: Looks good!

File: django/contrib/admin/static/admin/css/base.css [hunk 8]
```diff
@@ -587,7 +587,7 @@ input[type=button][disabled].default {
     font-weight: 400;
     font-size: 0.8125rem;
     text-align: left;
-    background: var(--primary);
+    background: var(--header-bg);
```
Comment (thibaudcolas) [inline django/contrib/admin/static/admin/css/base.css:590]: Looks good.

File: django/contrib/admin/static/admin/css/base.css [hunk 9]
```diff
@@ -44,14 +44,14 @@ html[data-theme="light"],
     --button-fg: #fff;
     --button-bg: var(--primary);
     --button-hover-bg: #609ab6;
-    --default-button-bg: var(--secondary);
+    --default-button-bg: var(--button-fg);
```
Comment (thibaudcolas) [inline django/contrib/admin/static/admin/css/base.css:47]: This effectively makes primary buttons harder to spot than secondary buttons so isn’t the right way forward.

File: django/contrib/admin/static/admin/css/base.css [hunk 10]
```diff
@@ -44,14 +44,14 @@ html[data-theme="light"],
...
     --default-button-hover-bg: #205067;
     --close-button-bg: #747474;
     --close-button-hover-bg: #333;
     --delete-button-bg: #ba2121;
     --delete-button-hover-bg: #a41515;
 
-    --object-tools-fg: var(--button-fg);
+    --object-tools-fg: var(--primary-fg);
```
Comment (thibaudcolas) [inline django/contrib/admin/static/admin/css/base.css:54]: As far as I can see this does nothing? Both colors are #fff.

File: django/contrib/admin/static/admin/css/base.css [hunk 11]
```diff
@@ -518,7 +518,7 @@ select[multiple] {
     padding: 10px 15px;
     border: none;
     border-radius: 4px;
-    color: var(--button-fg);
+    color: var(--body-fg);
```
Comment (thibaudcolas) [inline django/contrib/admin/static/admin/css/base.css:521]: This unfortunately doesn’t seem clear enough to me. It technically passes WCAG contrast checks but is still in my opinion too low contrast for such a critical component, and also as-is makes those buttons more prominent than primary form buttons.

File: django/contrib/admin/static/admin/css/base.css [hunk 12]
```diff
@@ -21,10 +21,10 @@ html[data-theme="light"],
     --header-link-color: var(--primary-fg);
 
     --breadcrumbs-fg: #c4dce8;
-    --breadcrumbs-link-fg: var(--body-bg);
+    --breadcrumbs-link-fg: var(--body-loud-color);
```
Comment (thibaudcolas) [inline django/contrib/admin/static/admin/css/base.css:24]: This is better but misses the fact breadcrumb links are still too low-contrast on hover. Perhaps we could proceed with this as-is though.

For hover, my preference would be to simply underline the links so color is no longer an issue.

File: django/contrib/admin/static/admin/css/base.css [hunk 13]
```diff
@@ -704,7 +704,7 @@ div.breadcrumbs {
     background: var(--breadcrumbs-bg);
     padding: 10px 40px;
     border: none;
-    color: var(--breadcrumbs-fg);
+    color: var(--body-fg);
```
Comment (thibaudcolas) [inline django/contrib/admin/static/admin/css/base.css:707]: Looking good!

**Review by thibaudcolas — COMMENTED** [review]
Thank you @nimra200! Some of this work seems ready to go, some needs further iterations.

**Comment by felixxm** [comment]
@nimra200 Do you have time to keep working on this?

**Comment by nimra200** [comment]
Unfortunately I do not have time to keep working on this.

________________________________

**Pull request closed without merging**